Sources/ArcGISToolkit/Components/FormView/FormInputs/FormInputModel.swift

```diff
@@ -34,67 +34,50 @@ class FormInputModel: ObservableObject {
     
     private var element: FieldFormElement
     
-    private var tasks = [Task<Void, Never>]()
-    
+    private var observationTask: Task<Void, Never>?
+
     deinit {
-        clearTasks()
+        observationTask?.cancel()
     }
     
     /// Initializes a form input model.
-    public init(fieldFormElement: FieldFormElement) {
+    init(fieldFormElement: FieldFormElement) {
         element = fieldFormElement
         isRequired = element.isRequired
         isEditable = element.isEditable
         value = element.value
         formattedValue = element.formattedValue
         
         // Kick off tasks to monitor required, editable and value.
-        tasks.append(
-            contentsOf: [
-                observeIsRequiredTask,
-                observeIsEditableTask,
-                observeValueTask
-            ]
-        )
-    }
-    
-    /// Cancels and removes tasks.
-    private func clearTasks() {
-        tasks.forEach { task in
-            task.cancel()
-        }
-        tasks.removeAll()
-    }
-    
-    /// A detached task observing changes in the required state.
-    private var observeIsRequiredTask: Task<Void, Never> {
-        Task.detached { [unowned self] in
-            for await isRequired in element.$isRequired {
-                await MainActor.run {
-                    self.isRequired = isRequired
+        observationTask = Task {
```

Comment:
Should this be a detached task? Otherwise the task is implicitly holding onto self strongly, preventing it from the model from going away and the task from being cancelled.